File: app/src/main/java/com/example/mytask/ViewListAdapter.java
package com.example.mytask;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class ViewListAdapter extends RecyclerView.Adapter<ViewListAdapter.ViewHolder>{
    private List<Productfile>productlist = new ArrayList<>();
    final List templist = new ArrayList<>();
    private Context context;
    int lastPosition = 0;
    SessionManager sessionManager;



    public ViewListAdapter(Context context, List<Productfile> productList) {
        this.context = context;
        this.productlist = productList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(context).inflate(R.layout.view_item_view, parent, false);

        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        final Productfile productfile=productlist.get(position);
        sessionManager=new SessionManager();
//         holder.Message.setVisibility(View.VISIBLE);
        if (productfile.getCount()!=null)
        if (productfile.getCount()==0){
            holder.itemname.setText(productfile.getName());
            holder.plist_text.setText(productfile.getDescription());
            holder.tv_quantity.setText(String.valueOf(productfile.getCount()));
            holder.price.setText("Rs "+String.valueOf(productfile.getPrice()));
            holder.tv_quantity.setVisibility(View.GONE);
            holder.Add.setVisibility(View.VISIBLE);
            holder.cart_minus_img.setVisibility(View.GONE);
            holder.cart_plus_img.setVisibility(View.GONE);
        }else{
            holder.itemname.setText(productfile.getName());
            holder.plist_text.setText(productfile.getDescription());
            holder.tv_quantity.setText(String.valueOf(productfile.getCount()));
            holder.price.setText("Rs "+String.valueOf(productfile.getPrice()));
            holder.tv_quantity.setVisibility(View.VISIBLE);
            holder.Add.setVisibility(View.GONE);
            holder.cart_minus_img.setVisibility(View.VISIBLE);
            holder.cart_plus_img.setVisibility(View.VISIBLE);
        }
        holder.Add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                holder.tv_quantity.setVisibility(View.VISIBLE);
                holder.Add.setVisibility(View.GONE);
                holder.cart_minus_img.setVisibility(View.VISIBLE);
                holder.cart_plus_img.setVisibility(View.VISIBLE);
                holder.tv_quantity.setText(String.valueOf(1));
                ((ViewCart) context).AddcartValue("1","Add",productfile);
            }
        });
        holder.cart_plus_img.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
           String total=  holder.tv_quantity.getText().toString();
                int quanitiy = Integer.parseInt(total);
            if (!(Integer.parseInt(total) ==20)){
               quanitiy=Integer.parseInt(total)+1;
                 holder.tv_quantity.setText(String.valueOf(quanitiy));

                ((ViewCart) context).AddcartValue(String.valueOf(quanitiy),"Add", productfile);
            }


            }
        });
        holder.cart_minus_img.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String total=     holder.tv_quantity.getText().toString();
                int quanitiy = Integer.parseInt(total);
                if (Integer.parseInt(total)==1){
                    quanitiy =0;
                    holder.tv_quantity.setVisibility(View.GONE);
                    holder.Add.setVisibility(View.VISIBLE);
                    holder.cart_minus_img.setVisibility(View.GONE);
                    holder.cart_plus_img.setVisibility(View.GONE);
                    ((ViewCart) context).AddcartValue(String.valueOf(quanitiy),"minus", productfile);
                }else{
                   quanitiy=Integer.parseInt(total)-1;
                    holder.tv_quantity.setText(String.valueOf(quanitiy));
                    ((ViewCart) context).AddcartValue(String.valueOf(quanitiy),"minus", productfile);
                }

            }
        });


    }


    @Override
    public int getItemCount() {
        if (productlist != null) {
            return productlist.size();
        }
        return 0;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
         TextView plist_text, itemname, Add, tv_quantity,price;
        ImageView cart_minus_img, cart_plus_img, Message;


        public ViewHolder(View itemView) {
            super(itemView);
            cart_minus_img = itemView.findViewById(R.id.cart_minus_img);
            cart_plus_img = itemView.findViewById(R.id.cart_plus_img);
            itemname = itemView.findViewById(R.id.from_name);
            plist_text = itemView.findViewById(R.id.plist_text);
            Add = itemView.findViewById(R.id.Add);
            tv_quantity = itemView.findViewById(R.id.cart_quantity_tv);
            Message = itemView.findViewById(R.id.Message);
            price = itemView.findViewById(R.id.price);

        }
    }
}

File: app/src/main/java/com/example/mytask/adapters/ViewListAdapter.kt
package com.example.mytask.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.mytask.R
import com.example.mytask.ViewCart
import com.example.mytask.model.Productfile
import java.util.*

class ViewListAdapter(private val context: Context, productList: List<Productfile>?) : RecyclerView.Adapter<ViewListAdapter.ViewHolder>() {
    private var productlist: List<Productfile>? = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(context).inflate(R.layout.view_item_view, parent, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val productFile = productlist!![position]
        if (productFile.count != null) if (productFile.count == 0) {
            holder.itemName.text = productFile.name
            holder.plistText.text = productFile.description
            holder.tvQuantity.text = productFile.count.toString()
            holder.price.text = "Rs " + productFile.price.toString()
            holder.tvQuantity.visibility = View.GONE
            holder.add.visibility = View.VISIBLE
            holder.cartMinusImg.visibility = View.GONE
            holder.cartPlusImg.visibility = View.GONE
        } else {
            holder.itemName.text = productFile.name
            holder.plistText.text = productFile.description
            holder.tvQuantity.text = productFile.count.toString()
            holder.price.text = "Rs " + productFile.price.toString()
            holder.tvQuantity.visibility = View.VISIBLE
            holder.add.visibility = View.GONE
            holder.cartMinusImg.visibility = View.VISIBLE
            holder.cartPlusImg.visibility = View.VISIBLE
        }
        holder.add.setOnClickListener {
            holder.tvQuantity.visibility = View.VISIBLE
            holder.add.visibility = View.GONE
            holder.cartMinusImg.visibility = View.VISIBLE
            holder.cartPlusImg.visibility = View.VISIBLE
            holder.tvQuantity.text = 1.toString()
            (context as ViewCart).addcartValue("1", "Add", productFile)
        }
        holder.cartPlusImg.setOnClickListener {
            val total = holder.tvQuantity.text.toString()
            var quanitiy = total.toInt()
            if (total.toInt() != 20) {
                quanitiy = total.toInt() + 1
                holder.tvQuantity.text = quanitiy.toString()
                (context as ViewCart).addcartValue(quanitiy.toString(), "Add", productFile)
            }
        }
        holder.cartMinusImg.setOnClickListener {
            val total = holder.tvQuantity.text.toString()
            var quanitiy = total.toInt()
            if (total.toInt() == 1) {
                quanitiy = 0
                holder.tvQuantity.visibility = View.GONE
                holder.add.visibility = View.VISIBLE
                holder.cartMinusImg.visibility = View.GONE
                holder.cartPlusImg.visibility = View.GONE
                (context as ViewCart).addcartValue(quanitiy.toString(), "minus", productFile)
            } else {
                quanitiy = total.toInt() - 1
                holder.tvQuantity.text = quanitiy.toString()
                (context as ViewCart).addcartValue(quanitiy.toString(), "minus", productFile)
            }
        }
    }

    override fun getItemCount(): Int {
        return productlist?.size ?: 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var plistText = itemView.findViewById<TextView>(R.id.plist_text)
        var itemName = itemView.findViewById<TextView>(R.id.from_name)
        var add = itemView.findViewById<TextView>(R.id.Add)
        var tvQuantity = itemView.findViewById<TextView>(R.id.cart_quantity_tv)
        var price = itemView.findViewById<TextView>(R.id.price)
        var cartMinusImg = itemView.findViewById<ImageView>(R.id.cart_minus_img)
        var cartPlusImg = itemView.findViewById<ImageView>(R.id.cart_plus_img)


    }

    init {
        productlist = productList
    }
}